Src/Microsoft.Dynamic/Actions/ReflectedPropertyTracker.cs

```diff
@@ -80,11 +74,7 @@ public class ReflectedPropertyTracker : PropertyTracker {
             return _propInfo.GetIndexParameters();
         }
 
-        public PropertyInfo Property {
-            get {
-                return _propInfo;
-            }
-        }
+        public PropertyInfo Property => _propInfo;
```

Comment:
The current master and current PR will have closer IL code (only one difference because of readonly).
The proposed changes will have more differences and a couple extra lines (because of the property inside a property).

Current master

` .class public auto ansi beforefieldinit Microsoft.Scripting.Actions.ReflectedPropertyTracker
	extends Microsoft.Scripting.Actions.PropertyTracker
{
	// Fields
	.field private class [mscorlib]System.Reflection.PropertyInfo _propInfo

	// Methods
	.method public hidebysig specialname rtspecialname 
		instance void .ctor (
			class [mscorlib]System.Reflection.PropertyInfo 'property'
		) cil managed 
	{
		// Method begins at RVA 0x567a8
		// Code size 14 (0xe)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance void Microsoft.Scripting.Actions.PropertyTracker::.ctor()
		IL_0006: ldarg.0
		IL_0007: ldarg.1
		IL_0008: stfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_000d: ret
	} // end of method ReflectedPropertyTracker::.ctor

	.method public hidebysig specialname virtual 
		instance string get_Name () cil managed 
	{
		// Method begins at RVA 0x567b7
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance string [mscorlib]System.Reflection.MemberInfo::get_Name()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_Name

	.method public hidebysig specialname virtual 
		instance class [mscorlib]System.Type get_DeclaringType () cil managed 
	{
		// Method begins at RVA 0x567c4
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.MemberInfo::get_DeclaringType()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_DeclaringType

	.method public hidebysig specialname virtual 
		instance bool get_IsStatic () cil managed 
	{
		// Method begins at RVA 0x567d1
		// Code size 24 (0x18)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldc.i4.1
		IL_0002: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetGetMethod(bool)
		IL_0007: dup
		IL_0008: brtrue.s IL_0012

		IL_000a: pop
		IL_000b: ldarg.0
		IL_000c: ldc.i4.1
		IL_000d: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetSetMethod(bool)

		IL_0012: callvirt instance bool [mscorlib]System.Reflection.MethodBase::get_IsStatic()
		IL_0017: ret
	} // end of method ReflectedPropertyTracker::get_IsStatic

	.method public hidebysig specialname virtual 
		instance class [mscorlib]System.Type get_PropertyType () cil managed 
	{
		// Method begins at RVA 0x567ea
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.PropertyInfo::get_PropertyType()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_PropertyType

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetGetMethod () cil managed 
	{
		// Method begins at RVA 0x567f7
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetGetMethod()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetGetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetSetMethod () cil managed 
	{
		// Method begins at RVA 0x56804
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetSetMethod()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetSetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetGetMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x56811
		// Code size 13 (0xd)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: ldarg.1
		IL_0007: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetGetMethod(bool)
		IL_000c: ret
	} // end of method ReflectedPropertyTracker::GetGetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetSetMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x5681f
		// Code size 13 (0xd)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: ldarg.1
		IL_0007: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetSetMethod(bool)
		IL_000c: ret
	} // end of method ReflectedPropertyTracker::GetSetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetDeleteMethod () cil managed 
	{
		// Method begins at RVA 0x5682d
		// Code size 8 (0x8)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldc.i4.0
		IL_0002: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetDeleteMethod(bool)
		IL_0007: ret
	} // end of method ReflectedPropertyTracker::GetDeleteMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetDeleteMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x56838
		// Code size 93 (0x5d)
		.maxstack 4
		.locals init (
			[0] class [mscorlib]System.Reflection.MethodInfo res
		)

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.MemberInfo::get_DeclaringType()
		IL_000b: ldstr "Delete"
		IL_0010: ldarg.0
		IL_0011: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0016: callvirt instance string [mscorlib]System.Reflection.MemberInfo::get_Name()
		IL_001b: call string [mscorlib]System.String::Concat(string, string)
		IL_0020: ldarg.1
		IL_0021: brtrue.s IL_0026

		IL_0023: ldc.i4.0
		IL_0024: br.s IL_0028

		IL_0026: ldc.i4.s 32

		IL_0028: ldc.i4.8
		IL_0029: or
		IL_002a: ldc.i4.4
		IL_002b: or
		IL_002c: ldc.i4.s 16
		IL_002e: or
		IL_002f: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Type::GetMethod(string, valuetype [mscorlib]System.Reflection.BindingFlags)
		IL_0034: stloc.0
		IL_0035: ldloc.0
		IL_0036: ldnull
		IL_0037: call bool [mscorlib]System.Reflection.MethodInfo::op_Inequality(class [mscorlib]System.Reflection.MethodInfo, class [mscorlib]System.Reflection.MethodInfo)
		IL_003c: brfalse.s IL_005b

		IL_003e: ldloc.0
		IL_003f: callvirt instance bool [mscorlib]System.Reflection.MethodBase::get_IsSpecialName()
		IL_0044: brfalse.s IL_005b

		IL_0046: ldloc.0
		IL_0047: ldtoken Microsoft.Scripting.Runtime.PropertyMethodAttribute
		IL_004c: call class [mscorlib]System.Type [mscorlib]System.Type::GetTypeFromHandle(valuetype [mscorlib]System.RuntimeTypeHandle)
		IL_0051: ldc.i4.1
		IL_0052: callvirt instance bool [mscorlib]System.Reflection.MemberInfo::IsDefined(class [mscorlib]System.Type, bool)
		IL_0057: brfalse.s IL_005b

		IL_0059: ldloc.0
		IL_005a: ret

		IL_005b: ldnull
		IL_005c: ret
	} // end of method ReflectedPropertyTracker::GetDeleteMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.ParameterInfo[] GetIndexParameters () cil managed 
	{
		// Method begins at RVA 0x568a1
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Reflection.ParameterInfo[] [mscorlib]System.Reflection.PropertyInfo::GetIndexParameters()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetIndexParameters

	.method public hidebysig specialname 
		instance class [mscorlib]System.Reflection.PropertyInfo get_Property () cil managed 
	{
		// Method begins at RVA 0x568ae
		// Code size 7 (0x7)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: ret
	} // end of method ReflectedPropertyTracker::get_Property

	.method public hidebysig virtual 
		instance string ToString () cil managed 
	{
		// Method begins at RVA 0x568b6
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance string [mscorlib]System.Object::ToString()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::ToString

	// Properties
	.property instance string Name()
	{
		.get instance string Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Name()
	}
	.property instance class [mscorlib]System.Type DeclaringType()
	{
		.get instance class [mscorlib]System.Type Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_DeclaringType()
	}
	.property instance bool IsStatic()
	{
		.get instance bool Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_IsStatic()
	}
	.property instance class [mscorlib]System.Type PropertyType()
	{
		.get instance class [mscorlib]System.Type Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_PropertyType()
	}
	.property instance class [mscorlib]System.Reflection.PropertyInfo Property()
	{
		.get instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
	}

} // end of class Microsoft.Scripting.Actions.ReflectedPropertyTracker`

Current PR

`
.class public auto ansi beforefieldinit Microsoft.Scripting.Actions.ReflectedPropertyTracker
	extends Microsoft.Scripting.Actions.PropertyTracker
{
	// Fields
	.field private initonly class [mscorlib]System.Reflection.PropertyInfo _propInfo

	// Methods
	.method public hidebysig specialname rtspecialname 
		instance void .ctor (
			class [mscorlib]System.Reflection.PropertyInfo 'property'
		) cil managed 
	{
		// Method begins at RVA 0x567a8
		// Code size 14 (0xe)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance void Microsoft.Scripting.Actions.PropertyTracker::.ctor()
		IL_0006: ldarg.0
		IL_0007: ldarg.1
		IL_0008: stfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_000d: ret
	} // end of method ReflectedPropertyTracker::.ctor

	.method public hidebysig specialname virtual 
		instance string get_Name () cil managed 
	{
		// Method begins at RVA 0x567b7
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance string [mscorlib]System.Reflection.MemberInfo::get_Name()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_Name

	.method public hidebysig specialname virtual 
		instance class [mscorlib]System.Type get_DeclaringType () cil managed 
	{
		// Method begins at RVA 0x567c4
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.MemberInfo::get_DeclaringType()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_DeclaringType

	.method public hidebysig specialname virtual 
		instance bool get_IsStatic () cil managed 
	{
		// Method begins at RVA 0x567d1
		// Code size 24 (0x18)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldc.i4.1
		IL_0002: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetGetMethod(bool)
		IL_0007: dup
		IL_0008: brtrue.s IL_0012

		IL_000a: pop
		IL_000b: ldarg.0
		IL_000c: ldc.i4.1
		IL_000d: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetSetMethod(bool)

		IL_0012: callvirt instance bool [mscorlib]System.Reflection.MethodBase::get_IsStatic()
		IL_0017: ret
	} // end of method ReflectedPropertyTracker::get_IsStatic

	.method public hidebysig specialname virtual 
		instance class [mscorlib]System.Type get_PropertyType () cil managed 
	{
		// Method begins at RVA 0x567ea
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.PropertyInfo::get_PropertyType()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_PropertyType

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetGetMethod () cil managed 
	{
		// Method begins at RVA 0x567f7
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetGetMethod()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetGetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetSetMethod () cil managed 
	{
		// Method begins at RVA 0x56804
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetSetMethod()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetSetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetGetMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x56811
		// Code size 13 (0xd)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: ldarg.1
		IL_0007: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetGetMethod(bool)
		IL_000c: ret
	} // end of method ReflectedPropertyTracker::GetGetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetSetMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x5681f
		// Code size 13 (0xd)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: ldarg.1
		IL_0007: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetSetMethod(bool)
		IL_000c: ret
	} // end of method ReflectedPropertyTracker::GetSetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetDeleteMethod () cil managed 
	{
		// Method begins at RVA 0x5682d
		// Code size 8 (0x8)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldc.i4.0
		IL_0002: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetDeleteMethod(bool)
		IL_0007: ret
	} // end of method ReflectedPropertyTracker::GetDeleteMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetDeleteMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x56838
		// Code size 93 (0x5d)
		.maxstack 4
		.locals init (
			[0] class [mscorlib]System.Reflection.MethodInfo res
		)

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.MemberInfo::get_DeclaringType()
		IL_000b: ldstr "Delete"
		IL_0010: ldarg.0
		IL_0011: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0016: callvirt instance string [mscorlib]System.Reflection.MemberInfo::get_Name()
		IL_001b: call string [mscorlib]System.String::Concat(string, string)
		IL_0020: ldarg.1
		IL_0021: brtrue.s IL_0026

		IL_0023: ldc.i4.0
		IL_0024: br.s IL_0028

		IL_0026: ldc.i4.s 32

		IL_0028: ldc.i4.8
		IL_0029: or
		IL_002a: ldc.i4.4
		IL_002b: or
		IL_002c: ldc.i4.s 16
		IL_002e: or
		IL_002f: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Type::GetMethod(string, valuetype [mscorlib]System.Reflection.BindingFlags)
		IL_0034: stloc.0
		IL_0035: ldloc.0
		IL_0036: ldnull
		IL_0037: call bool [mscorlib]System.Reflection.MethodInfo::op_Inequality(class [mscorlib]System.Reflection.MethodInfo, class [mscorlib]System.Reflection.MethodInfo)
		IL_003c: brfalse.s IL_005b

		IL_003e: ldloc.0
		IL_003f: callvirt instance bool [mscorlib]System.Reflection.MethodBase::get_IsSpecialName()
		IL_0044: brfalse.s IL_005b

		IL_0046: ldloc.0
		IL_0047: ldtoken Microsoft.Scripting.Runtime.PropertyMethodAttribute
		IL_004c: call class [mscorlib]System.Type [mscorlib]System.Type::GetTypeFromHandle(valuetype [mscorlib]System.RuntimeTypeHandle)
		IL_0051: ldc.i4.1
		IL_0052: callvirt instance bool [mscorlib]System.Reflection.MemberInfo::IsDefined(class [mscorlib]System.Type, bool)
		IL_0057: brfalse.s IL_005b

		IL_0059: ldloc.0
		IL_005a: ret

		IL_005b: ldnull
		IL_005c: ret
	} // end of method ReflectedPropertyTracker::GetDeleteMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.ParameterInfo[] GetIndexParameters () cil managed 
	{
		// Method begins at RVA 0x568a1
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance class [mscorlib]System.Reflection.ParameterInfo[] [mscorlib]System.Reflection.PropertyInfo::GetIndexParameters()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetIndexParameters

	.method public hidebysig specialname 
		instance class [mscorlib]System.Reflection.PropertyInfo get_Property () cil managed 
	{
		// Method begins at RVA 0x568ae
		// Code size 7 (0x7)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: ret
	} // end of method ReflectedPropertyTracker::get_Property

	.method public hidebysig virtual 
		instance string ToString () cil managed 
	{
		// Method begins at RVA 0x568b6
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::_propInfo
		IL_0006: callvirt instance string [mscorlib]System.Object::ToString()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::ToString

	// Properties
	.property instance string Name()
	{
		.get instance string Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Name()
	}
	.property instance class [mscorlib]System.Type DeclaringType()
	{
		.get instance class [mscorlib]System.Type Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_DeclaringType()
	}
	.property instance bool IsStatic()
	{
		.get instance bool Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_IsStatic()
	}
	.property instance class [mscorlib]System.Type PropertyType()
	{
		.get instance class [mscorlib]System.Type Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_PropertyType()
	}
	.property instance class [mscorlib]System.Reflection.PropertyInfo Property()
	{
		.get instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
	}

} // end of class Microsoft.Scripting.Actions.ReflectedPropertyTracker
`

Proposed changes

`
.class public auto ansi beforefieldinit Microsoft.Scripting.Actions.ReflectedPropertyTracker
	extends Microsoft.Scripting.Actions.PropertyTracker
{
	// Fields
	.field private initonly class [mscorlib]System.Reflection.PropertyInfo '<Property>k__BackingField'
	.custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
		01 00 00 00
	)

	// Methods
	.method public hidebysig specialname rtspecialname 
		instance void .ctor (
			class [mscorlib]System.Reflection.PropertyInfo 'property'
		) cil managed 
	{
		// Method begins at RVA 0x567a8
		// Code size 14 (0xe)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance void Microsoft.Scripting.Actions.PropertyTracker::.ctor()
		IL_0006: ldarg.0
		IL_0007: ldarg.1
		IL_0008: stfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::'<Property>k__BackingField'
		IL_000d: ret
	} // end of method ReflectedPropertyTracker::.ctor

	.method public hidebysig specialname virtual 
		instance string get_Name () cil managed 
	{
		// Method begins at RVA 0x567b7
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance string [mscorlib]System.Reflection.MemberInfo::get_Name()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_Name

	.method public hidebysig specialname virtual 
		instance class [mscorlib]System.Type get_DeclaringType () cil managed 
	{
		// Method begins at RVA 0x567c4
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.MemberInfo::get_DeclaringType()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_DeclaringType

	.method public hidebysig specialname virtual 
		instance bool get_IsStatic () cil managed 
	{
		// Method begins at RVA 0x567d1
		// Code size 24 (0x18)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldc.i4.1
		IL_0002: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetGetMethod(bool)
		IL_0007: dup
		IL_0008: brtrue.s IL_0012

		IL_000a: pop
		IL_000b: ldarg.0
		IL_000c: ldc.i4.1
		IL_000d: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetSetMethod(bool)

		IL_0012: callvirt instance bool [mscorlib]System.Reflection.MethodBase::get_IsStatic()
		IL_0017: ret
	} // end of method ReflectedPropertyTracker::get_IsStatic

	.method public hidebysig specialname virtual 
		instance class [mscorlib]System.Type get_PropertyType () cil managed 
	{
		// Method begins at RVA 0x567ea
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.PropertyInfo::get_PropertyType()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::get_PropertyType

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetGetMethod () cil managed 
	{
		// Method begins at RVA 0x567f7
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetGetMethod()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetGetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetSetMethod () cil managed 
	{
		// Method begins at RVA 0x56804
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetSetMethod()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetSetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetGetMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x56811
		// Code size 13 (0xd)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: ldarg.1
		IL_0007: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetGetMethod(bool)
		IL_000c: ret
	} // end of method ReflectedPropertyTracker::GetGetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetSetMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x5681f
		// Code size 13 (0xd)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: ldarg.1
		IL_0007: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Reflection.PropertyInfo::GetSetMethod(bool)
		IL_000c: ret
	} // end of method ReflectedPropertyTracker::GetSetMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetDeleteMethod () cil managed 
	{
		// Method begins at RVA 0x5682d
		// Code size 8 (0x8)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldc.i4.0
		IL_0002: callvirt instance class [mscorlib]System.Reflection.MethodInfo Microsoft.Scripting.Actions.PropertyTracker::GetDeleteMethod(bool)
		IL_0007: ret
	} // end of method ReflectedPropertyTracker::GetDeleteMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.MethodInfo GetDeleteMethod (
			bool privateMembers
		) cil managed 
	{
		// Method begins at RVA 0x56838
		// Code size 93 (0x5d)
		.maxstack 4
		.locals init (
			[0] class [mscorlib]System.Reflection.MethodInfo res
		)

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance class [mscorlib]System.Type [mscorlib]System.Reflection.MemberInfo::get_DeclaringType()
		IL_000b: ldstr "Delete"
		IL_0010: ldarg.0
		IL_0011: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0016: callvirt instance string [mscorlib]System.Reflection.MemberInfo::get_Name()
		IL_001b: call string [mscorlib]System.String::Concat(string, string)
		IL_0020: ldarg.1
		IL_0021: brtrue.s IL_0026

		IL_0023: ldc.i4.0
		IL_0024: br.s IL_0028

		IL_0026: ldc.i4.s 32

		IL_0028: ldc.i4.8
		IL_0029: or
		IL_002a: ldc.i4.4
		IL_002b: or
		IL_002c: ldc.i4.s 16
		IL_002e: or
		IL_002f: callvirt instance class [mscorlib]System.Reflection.MethodInfo [mscorlib]System.Type::GetMethod(string, valuetype [mscorlib]System.Reflection.BindingFlags)
		IL_0034: stloc.0
		IL_0035: ldloc.0
		IL_0036: ldnull
		IL_0037: call bool [mscorlib]System.Reflection.MethodInfo::op_Inequality(class [mscorlib]System.Reflection.MethodInfo, class [mscorlib]System.Reflection.MethodInfo)
		IL_003c: brfalse.s IL_005b

		IL_003e: ldloc.0
		IL_003f: callvirt instance bool [mscorlib]System.Reflection.MethodBase::get_IsSpecialName()
		IL_0044: brfalse.s IL_005b

		IL_0046: ldloc.0
		IL_0047: ldtoken Microsoft.Scripting.Runtime.PropertyMethodAttribute
		IL_004c: call class [mscorlib]System.Type [mscorlib]System.Type::GetTypeFromHandle(valuetype [mscorlib]System.RuntimeTypeHandle)
		IL_0051: ldc.i4.1
		IL_0052: callvirt instance bool [mscorlib]System.Reflection.MemberInfo::IsDefined(class [mscorlib]System.Type, bool)
		IL_0057: brfalse.s IL_005b

		IL_0059: ldloc.0
		IL_005a: ret

		IL_005b: ldnull
		IL_005c: ret
	} // end of method ReflectedPropertyTracker::GetDeleteMethod

	.method public hidebysig virtual 
		instance class [mscorlib]System.Reflection.ParameterInfo[] GetIndexParameters () cil managed 
	{
		// Method begins at RVA 0x568a1
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance class [mscorlib]System.Reflection.ParameterInfo[] [mscorlib]System.Reflection.PropertyInfo::GetIndexParameters()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::GetIndexParameters

	.method public hidebysig specialname 
		instance class [mscorlib]System.Reflection.PropertyInfo get_Property () cil managed 
	{
		.custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
			01 00 00 00
		)
		// Method begins at RVA 0x568ae
		// Code size 7 (0x7)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: ldfld class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::'<Property>k__BackingField'
		IL_0006: ret
	} // end of method ReflectedPropertyTracker::get_Property

	.method public hidebysig virtual 
		instance string ToString () cil managed 
	{
		// Method begins at RVA 0x568b6
		// Code size 12 (0xc)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
		IL_0006: callvirt instance string [mscorlib]System.Object::ToString()
		IL_000b: ret
	} // end of method ReflectedPropertyTracker::ToString

	// Properties
	.property instance string Name()
	{
		.get instance string Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Name()
	}
	.property instance class [mscorlib]System.Type DeclaringType()
	{
		.get instance class [mscorlib]System.Type Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_DeclaringType()
	}
	.property instance bool IsStatic()
	{
		.get instance bool Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_IsStatic()
	}
	.property instance class [mscorlib]System.Type PropertyType()
	{
		.get instance class [mscorlib]System.Type Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_PropertyType()
	}
	.property instance class [mscorlib]System.Reflection.PropertyInfo Property()
	{
		.get instance class [mscorlib]System.Reflection.PropertyInfo Microsoft.Scripting.Actions.ReflectedPropertyTracker::get_Property()
	}

} // end of class Microsoft.Scripting.Actions.ReflectedPropertyTracker

`